hazelcast/src/main/java/com/hazelcast/internal/diagnostics/DiagnosticsLogFile.java

```diff
@@ -107,36 +104,22 @@ private void renderStaticPlugins() throws IOException {
     }
 
     private void renderPlugin(DiagnosticsPlugin plugin) throws IOException {
-        logWriter.clean();
-        plugin.run(logWriter);
-        //bufferedWriter.append()
-        int desiredLength = charBuff.length;
-        int actualSize = logWriter.length();
-        while (desiredLength < actualSize) {
-            desiredLength *= 2;
-        }
+        logWriter.init(printWriter);
 
-        if (desiredLength != charBuff.length) {
-            charBuff = new char[desiredLength];
-        }
-
-        logWriter.copyInto(charBuff);
-        bufferedWriter.write(charBuff, 0, actualSize);
-        fileLength += actualSize;
+        plugin.run(logWriter);
     }
 
-    private BufferedWriter newWriter() throws FileNotFoundException {
+    private PrintWriter newWriter() throws FileNotFoundException {
         FileOutputStream fos = new FileOutputStream(file, true);
         CharsetEncoder encoder = Charset.forName("UTF-8").newEncoder();
-        return new BufferedWriter(new OutputStreamWriter(fos, encoder));
+        return new PrintWriter(new BufferedWriter(new OutputStreamWriter(fos, encoder), Short.MAX_VALUE));
```

Comment:
I always get a headache from those stream/writer classes and we have four nested instances here.

Do I get it right that this is a file writer construct with a fixed buffer of 32 MB? And every time that (internal) buffer is full, it's written to disk?

And it's enough to close the outer `PrintWriter` to release all inner streams and writers? The `BufferedWriter` doesn't seem to call close() on its `OutputStreamWriter`. Maybe it's not needed at all, but I'm always a bit careful with resource handling.